components/hab/src/command/studio/enter.rs

```diff
@@ -86,6 +86,23 @@ mod inner {
                     let ident = PackageIdent::from_str(&format!("{}/{}",
                                                                 super::STUDIO_PACKAGE_IDENT,
                                                                 version[0]))?;
+                    // TODO(SM): This is the minimum change needed to allow the studio to specify
+                    // its dependencies. This is a duplicate of the code in `hab pkg exec` and
+                    // should be refactored.
+                    let pkg_install = PackageInstall::load(&ident, None)?;
+                    let cmd_env = pkg_install.environment_for_command()?;
+                    for (key, value) in cmd_env.into_iter() {
+                        debug!("Setting: {}='{}'", key, value);
+                        env::set_var(key, value);
+                    }
+
+                    let mut display_args = super::STUDIO_CMD.to_string();
+                    for arg in args {
+                        display_args.push(' ');
+                        display_args.push_str(arg.to_string_lossy().as_ref());
+                    }
+                    debug!("Running: {}", display_args);
```

Comment:
This could be a note for the future refactorings, rather than a change for this PR, but this is a lot of code for just printing out what's going to be run. It could be done more concisely (e.g., with some help from [SliceConcatExt::join](https://doc.rust-lang.org/std/slice/trait.SliceConcatExt.html#tymethod.join)).

(As an aside, it's currently doing all that concatenation even if debug logging is disabled, making it wasted effort.)

Reading more of the surrounding code, it also seems out of place here, given that this block of code is just trying to figure out which Studio binary to run; the arguments only really get used later on.

Similarly, jamming the environment variables directly into the running environment here, rather than somehow conveying it out to the point where the command is actually run, strikes me as a bad idea. Sure, it ends up working out to the same thing in this case, since we `exec` to become the studio process, but messing with global state at a point so distant from the actual execution just seems like begging for bugs.

Reply:
> (As an aside, it's currently doing all that concatenation even if debug logging is disabled, making it wasted effort.)
> 
> Reading more of the surrounding code, it also seems out of place here, given that this block of code is just trying to figure out which Studio binary to run; the arguments only really get used later on.

Yup 😭  I couldn't see a low-impact way of doing this only when necessary, `studio/enter.rs` is need of a lot of cleanup.  Ideally we'd do that work up front, but this will unblock a lot of the improvements to our release process. I'm open to suggestions, but as I noted above I feel that work should be it's own PR and keep this isolated to changes necessary to make backline a runtime dependency.

Reply:
@smacfarlane Yup, I totally get the constraints you're operating under. Feel free to take my suggestions as todo items on that future PR.

I _would_ like for us to be able to tackle that PR in the very near future, however, just so we can not let bad code lie any longer than absolutely necessary.

Reply:
I had to abandon https://github.com/habitat-sh/habitat/pull/6347, but I think that direction would take us towards something cleaner and more maintainable.